Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Services/ApiGatewayRouteConfigService.cs

```
@@ -0,0 +1,98 @@
+        {
+            var key = entry.Key.ToString();
+            if (key is null)
+                continue;
+            if (!(key.Equals(Constants.LambdaConfigEnvironmentVariablePrefix) || 
+                key.StartsWith($"{Constants.LambdaConfigEnvironmentVariablePrefix}_")))
+                continue;
+
+            var jsonValue = entry.Value?.ToString();
+            if (string.IsNullOrEmpty(jsonValue))
+                continue;
+            try
+            {
+                var config = JsonSerializer.Deserialize<ApiGatewayRouteConfig>(jsonValue);
+                if (config != null)
+                {
+                    _routeConfigs.Add(config);
+                }
+            }
+            catch (Exception)
+            {
+                try
+                {
+                    var config = JsonSerializer.Deserialize<List<ApiGatewayRouteConfig>>(jsonValue);
+                    if (config != null)
+                    {
+                        _routeConfigs.AddRange(config);
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Console.WriteLine($"Error deserializing environment variable {key}: {ex.Message}");
+                }
+            }
+        }
+    }
+
+    /// <inheritdoc />
+    public ApiGatewayRouteConfig? GetRouteConfig(string httpMethod, string path)
+    {
+        foreach (var routeConfig in _routeConfigs)
+        {
+            var template = TemplateParser.Parse(routeConfig.Path);
+
+            var matcher = new TemplateMatcher(template, GetDefaults(template));
+
+            var routeValueDictionary = new RouteValueDictionary();
+            if (!matcher.TryMatch(path, routeValueDictionary))
+                continue;
+            
+            if (!routeConfig.HttpMethod.Equals(httpMethod, StringComparison.InvariantCultureIgnoreCase))
+                continue;
+
+            return routeConfig;
+        }
+
+        return null;
```

should we throw exception here instead of returning null? is there any use case where null would be valid?

---

From a separation of concerns perspective, this function is trying to match routes. If a route doesn't match, I don't want it to blow up. It should indicate that the route doesn't match, which I am using null for. It is the responsibility of the caller to throw an exception if needed.